tests/Writing/CopyMethodsTest.php

```diff
@@ -239,4 +239,38 @@ public function testCopyUpdateReferencesMultiValue()
 
         $this->assertCount(2, $references);
     }
+
+    /**
+     * Verifies that transport::copy actually copies binary data of children nodes
+     */
+    public function testCopyChildrenBinaryData()
+    {
+        $src = '/tests_write_manipulation_copy/testCopyChildrenBinaryData/srcNode';
+        $dst = '/tests_write_manipulation_copy/testCopyChildrenBinaryData/dstNode';
+
+
+        $this->ws->copy($src, $dst);
+        $this->session->refresh(true);
+        $srcChild = $this->session->getNode($src.'/data.bin');
+        $dstChild = $this->session->getNode($dst.'/data.bin');
+
+        $srcProp = $srcChild->getProperty('jcr:data');
+        $dstProp = $dstChild->getProperty('jcr:data');
+
+        $this->assertThat(true,
+            $this->logicalAnd(
+                $this->equalTo(\PHPCR\PropertyType::BINARY, $srcProp->getType()),
+                $this->equalTo(\PHPCR\PropertyType::BINARY, $dstProp->getType())
+            )
+        );
+
+        $srcBin = $srcProp->getBinary();
+        $dstBin = $dstProp->getBinary();
+
+        if (!is_resource($srcBin) || !is_resource($dstBin)) {
```

Comment:
its the same. at the first assertion that fails, the test aborts.
semantically your code and what i propose would do the same, but i think
my version is easier to read, and you see which binary was not a resource